src/foam/nanos/auth/UserPropertyAvailabilityService.js

```diff
@@ -40,13 +40,20 @@ foam.CLASS({
           throw new AuthorizationException();
         }
 
-        ArraySink select = (ArraySink) userDAO.inX(x).where(MLang.EQ(targetProperty.equals("userName") ? User.USER_NAME : User.EMAIL, value)).select(new ArraySink());
-        
+        ArraySink select = (ArraySink) userDAO.inX(x).where(
+          AND(
```

Comment:
If you only care if one exists, a find() would be more efficient than a select().

Reply:
@thepinkcacao find(predicate) will perform where(predicate).select().limit(1) for you.

Reply:
okay, thank you!